src/components/errors/error-modal.component.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { useMemo } from 'react';
+import { InlineNotification } from '@carbon/react';
+import { useTranslation } from 'react-i18next';
+import { fieldRequiredErrCode, fieldOutOfBoundErrCode } from '../../validators/form-validator';
+import type { ValidationResult } from '../../types';
+import styles from './error.scss';
+
+const ErrorModal: React.FC<{ errors: ValidationResult[] }> = ({ errors }) => {
+  const { t } = useTranslation();
+
+  const errorMessages = {};
+
+  errors.forEach((error) => {
+    if (error?.errCode === fieldRequiredErrCode && !errorMessages[fieldRequiredErrCode]) {
+      errorMessages[fieldRequiredErrCode] = t('nullMandatoryField', 'Please fill the required fields');
+    } else if (error?.errCode === fieldOutOfBoundErrCode && !errorMessages[fieldOutOfBoundErrCode]) {
+      errorMessages[fieldOutOfBoundErrCode] = t('valuesOutOfBound', 'Some of the values are out of bounds');
+    }
+  });
```

Comment:
I would move this whole thing into the `useMemo()` below. that way it only depends on the `errors` prop, and the only value we need React to be aware of is the `errorMessage`.

Reply:
Done